e2e/helpers/hooks_helper.js

```diff
@@ -1,10 +1,35 @@
 /* global process */
 
+const fs = require('fs');
+const { clearString, screenshotOutputFolder } = require('codeceptjs/lib/utils');
+
+function buildOutputFileName (test, extension) {
+  let fileName = clearString(test.title);
+  if (test.ctx && test.ctx.test && test.ctx.test.type === 'hook') {
+    fileName += clearString(`_${test.ctx.test.title}`);
+  }
+  return screenshotOutputFolder(`${fileName}.${extension}`);
+}
+
 module.exports = class HooksHelpers extends Helper {
   _test(test) {
     test.retries(parseInt(process.env.TEST_RETRIES || '2'));
   }
 
+  async _failed(test) {
+    const logs = (await this.helpers['Puppeteer'].grabBrowserLogs()).map(log => {
+      return {
+        type: log.type(),
+        message: log.text(),
+        location: log.location().url,
+      };
+    });
+
+    if (logs.length > 0) {
+      fs.writeFileSync(buildOutputFileName(test, 'browser.log'), JSON.stringify(logs, undefined, 2));
```

Comment:
(0) - super minor
Browser.log is not really extension is it? It's more like suffix, but doesn't matter.
Not sure about the JSON.stringify, could we extract 2 as the constant number of spaces to make it obvious what it means? I had the google, other people would need to google as well.